### PR TITLE
Debloat channels.cr into multiple files

### PR DESCRIPTION
**Pull request by syeopite**
Cherry picked from #2215 with a few modifications:
  - channel folder is renamed to channels
  - parsing for channel home and featured channels are removed due to
    lack of infrastructure from other commits

(cherry picked from commit 44d18b8e147b47ad06a54cc6fd08423d9f39074d)

### CODE REVIEW

**Comment by SamantazFox** [comment]
Can you please keep `produce_channel_playlists_url` around? The informations about the "playlists" protobuf object contained in this function is important!

**Comment by syeopite** [comment]
Done

**Review by SamantazFox — CHANGES_REQUESTED** [review]

File: src/invidious.cr [hunk 1]
```diff
@@ -27,6 +27,7 @@ require "compress/zip"
 require "protodec/utils"
 require "./invidious/helpers/*"
 require "./invidious/*"
+require "./invidious/channels/**"
```
Comment (SamantazFox) [inline src/invidious.cr:30]: As in `spec/helpers_spec.cr`, this should be a single star, to match all files in the `channels` directory, unless you plan to plan to add subfolders, in this case this is the spec file that needs to be fixed.

**Review by SamantazFox — APPROVED** [review]

**Comment by SamantazFox** [comment]
@syeopite I just realized that `country` and `links` elements in the `AboutChannel` struct aren't part of the original source code (probably some leftover of your other changes). Could you just remove those?

https://github.com/syeopite/invidious/blob/5bcb689804b214229967c955e9e132a66c131186/src/invidious/channels/about.cr#L2-L22

https://github.com/syeopite/invidious/blob/5bcb689804b214229967c955e9e132a66c131186/src/invidious/channels/about.cr#L141-L144

**Comment by syeopite** [comment]
Oops, somehow missed that during cherry-picking. Should be fixed now.

**Pull request merged**